### PR TITLE
DEX-982 Remove /site-settings/apiKeys API and limit flatfileKey to logged in users only

### PR DESCRIPTION
**Pull request by hwindsor**
- Get site settings to work with or without edm

  Part of the site settings code just always call edm whether or not the
  user is modifying edm values.  Change it so it only calls edm if edm
  values are modified.

- Change "mock" imports to "unittest.mock" in tests

  From python3, "mock" is in the standard library as "unittest.mock".

- Add environment variables for site settings api key defaults

  - `TRANSLOADIT_KEY`
  - `TRANSLOADIT_TEMPLATE_ID`
  - `TRANSLOADIT_SERVICE`
  - `GOOGLE_MAPS_API_KEY`
  - `FLATFILE_KEY`
  - `RECAPTCHA_PUBLIC_KEY`
  - `RECAPTCHA_SECRET_KEY`

- Remove /site-settings/apiKeys API and change more settings to private

  Settings used only by back end can be set to private.
  
  These settings won't be returned in the `/site-settings/main/block` API
  for non admin users.

- DEX-982 Limit flatfileKey site setting to logged in users only

  Add "permission" to `SiteSetting.HOUSTON_SETTINGS` so we can restrict
  who gets certain settings in `/api/v1/site-settings/main/block`.

- Show docker-compose logs if failed in testing CI

---

Use `/api/v1/site-settings/main/block` which returns settings that the current user has access to.

The front end PR is WildMeOrg/codex-frontend#370

### CODE REVIEW

**Comment by Emily-Ke** [comment]
I asked @brmscheiner to take a look at this. I don't think we need all of these keys. I think that we are only using flatfileKey, googleMapsApiKey, and sentryDsn.

**Comment by karenc** [comment]
Why are we creating another endpoint `/api/v1/site-settings/apiKeys` for this...?

I just implemented recaptcha on the front end https://github.com/WildMeOrg/codex-frontend/pull/349 and there's already a standard way to get all site settings using `/api/v1/site-settings/main/block` so if we just include those settings by making them public, isn't that more consistent?

Any API keys used by the front end can't really be not public anyway... can it?

**Comment by karenc** [comment]
I've been changing site settings to use a default environment variable like:

```python
        'recaptchaPublicKey': {
            'type': str,   
            'public': True,                                                   
            'default': lambda: current_app.config.get('RECAPTCHA_PUBLIC_KEY'),
        },                     
```

This is so we can set the site setting using `docker-compose.override.yml` environment variable, but use the database value if it's set.  I believe this is more convenient for deployment so we can set the site up without inserting rows into the database.  What do people think about it?

**Comment by karenc** [comment]
I'm going to take over DEX-982, I can either open a new PR or continue with this one, whichever works

**Comment by bluemellophone** [comment]
> I'm going to take over DEX-982, I can either open a new PR or continue with this one, whichever works

Please continue this one

File: app/modules/site_settings/models.py [hunk 1]
```diff
@@ -116,38 +117,33 @@ class SiteSetting(db.Model, Timestamp):
...
+            'public': True,
+            'default': lambda: current_app.config.get('SENTRY_DSN'),
         },
         'flatfileKey': {
             'type': str,
-            'public': False,
-            'default': '',
-            'isApiKey': True,
+            'public': True,
+            'permission': lambda: not current_user.is_anonymous,
```
Comment (hwindsor) [inline app/modules/site_settings/models.py:145]: Only a researcher should be able to do a bulk upload (and I think this is only neccessary for bulk upload) so this should be is_researcher

**Review by bluemellophone — APPROVED** [review]
LGTM, please address Howard's comment then it will be ready for merging

**Review by hwindsor — COMMENTED** [review]
LGTM

**Pull request merged**